### PR TITLE
Some updates to refund endpoint

### PR DESCRIPTION
**Pull request by Shadonra**

### CODE REVIEW

File: sale.go [hunk 1]
```diff
@@ -21,17 +21,35 @@ func (c *Client) GetSale(saleID string) (*Sale, error) {
 	return sale, nil
 }
 
+// DEPRECATED
```
Comment (roopakv) [inline sale.go]: Why deprecated?

**Review by roopakv — COMMENTED** [review]
a little confused on Refund vs RefundSale. could you explain :)

File: types.go [hunk 2]
```diff
@@ -191,6 +191,12 @@ type (
 		FinalCapture   bool   `json:"final_capture,omitempty"`
 	}
 
+	RefundRequest struct {
+		Amount      PurchaseUnitAmount `json:"amount,omitempty"`
```
Comment (roopakv) [inline types.go]: nit: pointer

File: sale.go [hunk 3]
```diff
@@ -24,14 +24,10 @@ func (c *Client) GetSale(saleID string) (*Sale, error) {
 // RefundSale refunds a completed payment.
 // Use this call to refund a completed payment. Provide the sale_id in the URI and an empty JSON payload for a full refund. For partial refunds, you can include an amount.
 // Endpoint: POST /v2/payments/sale/ID/refund
-func (c *Client) RefundSale(saleID string, a *Amount) (*Refund, error) {
-	type refundRequest struct {
-		Amount *Amount `json:"amount"`
-	}
-
+func (c *Client) RefundSale(saleID string, r RefundRequest) (*Refund, error) {
```
Comment (roopakv) [inline sale.go:26]: oh I missed this, can this be a pointer too?

**Review by plutov — APPROVED** [review]

**Pull request merged**